src/handlers/actionsHandler.ts

```diff
@@ -109,12 +176,17 @@ export const dagAddressActions = async (
   try {
     const { address } = event.pathParameters || {};
 
+    const selectedTables = tableFilter(event).map(dagTable);
+
     return await paginatedQuery(
       extractPagination(event),
-      toCreatedAtOrdinalCursor,
-      fromCreatedAtOrdinalCursor,
-      { where: { source_addr: address, table_name: { in: dagActionsTables } }, 
-      orderBy: { created_at: 'desc' } },
+      hashCursor,
+      hashCursor,
+      {
+        where: { source_addr: address, table_name: { in: selectedTables } },
```

Comment:
We need to include destination addresses here for the transaction types that support it.